internal/install/recipe_installer.go

```diff
@@ -271,15 +273,25 @@ func (i *RecipeInstall) connectToPlatform() error {
 		loaderChan <- nil
 	}()
 
+	i.segment.Track("ConnectingToPlatformStarted")
 	i.progressIndicator.Start("Connecting to New Relic Platform")
+	start := time.Now()
 
 	loaded := <-loaderChan
 
+	durationMs := time.Since(start).Milliseconds()
+	ei := segment.NewEventInfo("ConnectingToPlatformComplete", "")
```

Comment:
A follow up item in a separate Jira could be adding/defining the new events (`ConnectingToPlatform*` and `ValidateNrql*`) to the `internal/install/types/errors.go` and even assessing, in case those events are only related to segment tracking, to extract those into a `types.go` file inside the new `segment` package you refactored.

Reply:
For sure, would be great to define these alongside the other events. It seemed simplest to me to set these up outside of the reporting system. There's probably more cleanup to do around segment logic built inside `install/recipes`. 

If it's not in a good state to you, I can address any of this in this PR, otherwise a follow-up PR could include a closer look at extracting everything else we can into `segment` and the event type cleanup you're suggesting

Reply:
I think following up in a separate ticket would be best for now.